[BUG] Recents activity crash on android 12

### Describe the bug

Recents activity crash

https://github.com/LawnchairLauncher/lawnchair/assets/51394647/9b319269-9806-4a21-9ae1-64203cc1f4fe



### Steps to reproduce

Steps to reproduce the behavior:
1. Call Recents
2. Close any application
4. See this error


### Expected behavior

Close apps without errors

### Screenshots

_No response_

### Device information

- Device: Lenovo Legion Y90
- OS: ZUI13, Android version 12.0
- App version: 14.Dev.(#1040) (14000202)


### Additional context

Lawnchair (Debug) bug report 30 июн. 2024 г. 16:25:01
version: 14.Dev.(#1040) (14000202)
commit: 82e3c1c
build.brand: Lenovo
build.device: diablo
build.display: SKQ1.211113.001 dev-keys
build.fingerprint: samsung/o1sxxx/o1s:11/RP1A.200720.012/G991BXXU3AUE1:user/release-keys
build.hardware: qcom
build.id: SKQ1.211113.001
build.manufacturer: Lenovo
build.model: Lenovo L71061
build.security.level: 2023-02-01
build.product: diablo
build.type: user
version.codename: REL
version.incremental: 13.5.870_230424
version.release: 12
version.sdk_int: 31
display.density_dpi: 480
isRecentsEnabled: true

error: Uncaught exception

java.lang.RuntimeException: Error creating RemoteTransitionCompatLaunchFromRecents
	at app.lawnchair.compatlib.twelve.QuickstepCompatFactoryVS.createRemoteTransition(QuickstepCompatFactoryVS.java:54)
	at app.lawnchair.compatlib.twelve.QuickstepCompatFactoryVS.$r8$lambda$3-ePWUsmTLiprJ9_ckxUmEESKwQ(Unknown Source:0)
	at app.lawnchair.compatlib.twelve.QuickstepCompatFactoryVS$$ExternalSyntheticLambda0.getRemoteTransition(D8$$SyntheticClass:0)
	at com.android.quickstep.RecentsActivity.getActivityLaunchOptions(RecentsActivity.java:282)
	at com.android.quickstep.views.TaskView.launchTaskAnimated(TaskView.java:868)
	at com.android.quickstep.views.TaskView.launchTasks(TaskView.java:1067)
	at com.android.quickstep.views.TaskView.onClick(TaskView.java:824)
	at com.android.quickstep.views.TaskView.$r8$lambda$E6ZJ14P3tv-M4UKAQSfwBJmmVow(Unknown Source:0)
	at com.android.quickstep.views.TaskView$$ExternalSyntheticLambda10.onClick(D8$$SyntheticClass:0)
	at android.view.View.performClick(View.java:7450)
	at android.view.View.performClickInternal(View.java:7427)
	at android.view.View.access$3700(View.java:835)
	at android.view.View$PerformClick.run(View.java:28702)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:225)
	at android.os.Looper.loop(Looper.java:326)
	at android.app.ActivityThread.main(ActivityThread.java:8151)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1013)
Caused by: java.lang.NoSuchMethodException: android.window.RemoteTransition.<init> [interface android.window.IRemoteTransition, interface android.app.IApplicationThread]
	at java.lang.Class.getConstructor0(Class.java:2363)
	at java.lang.Class.getConstructor(Class.java:1759)
	at app.lawnchair.compatlib.twelve.QuickstepCompatFactoryVS.createRemoteTransition(QuickstepCompatFactoryVS.java:46)
	... 20 more

--- a/compatLib/compatLibVS/src/main/java/app/lawnchair/compatlib/twelve/QuickstepCompatFactoryVS.java
+++ b/compatLib/compatLibVS/src/main/java/app/lawnchair/compatlib/twelve/QuickstepCompatFactoryVS.java
@@ -45,7 +45,8 @@ public class QuickstepCompatFactoryVS extends QuickstepCompatFactoryVR {
             Constructor<?> constructor =
                     remoteTransitionClass.getConstructor(
                             IRemoteTransition.class, IApplicationThread.class, String.class);
-            return (RemoteTransition) constructor.newInstance(remoteTransition, appThread, debugName);
+            return (RemoteTransition)
+                    constructor.newInstance(remoteTransition, appThread, debugName);
         } catch (ClassNotFoundException
                 | IllegalAccessException
                 | InstantiationException

--- a/compatLib/compatLibVS/src/main/java/app/lawnchair/compatlib/twelve/QuickstepCompatFactoryVS.java
+++ b/compatLib/compatLibVS/src/main/java/app/lawnchair/compatlib/twelve/QuickstepCompatFactoryVS.java
@@ -30,11 +30,7 @@ public class QuickstepCompatFactoryVS extends QuickstepCompatFactoryVR {
     @NonNull
     @Override
     public RemoteTransitionCompat getRemoteTransitionCompat() {
-        try {
-            return this::createRemoteTransition;
-        } catch (Throwable t) {
-            return super.getRemoteTransitionCompat();
-        }
+        return this::createRemoteTransition;
     }
 
     // TODO remove this as it causing glitches on first launch opening/closing app
@@ -44,15 +40,15 @@ public class QuickstepCompatFactoryVS extends QuickstepCompatFactoryVR {
             Class<?> remoteTransitionClass = Class.forName("android.window.RemoteTransition");
             Constructor<?> constructor =
                     remoteTransitionClass.getConstructor(
-                            IRemoteTransition.class, IApplicationThread.class, String.class);
-            return (RemoteTransition)
-                    constructor.newInstance(remoteTransition, appThread, debugName);
+                            IRemoteTransition.class, IApplicationThread.class);
+            return (RemoteTransition) constructor.newInstance(remoteTransition, appThread);
         } catch (ClassNotFoundException
                 | IllegalAccessException
                 | InstantiationException
                 | InvocationTargetException
                 | NoSuchMethodException e) {
-            throw new RuntimeException("Error creating RemoteTransitionCompat" + debugName, e);
+            return super.getRemoteTransitionCompat()
+                    .getRemoteTransition(remoteTransition, appThread, debugName);
         }
     }
 }

--- a/compatLib/compatLibVS/src/main/java/app/lawnchair/compatlib/twelve/QuickstepCompatFactoryVS.java
+++ b/compatLib/compatLibVS/src/main/java/app/lawnchair/compatlib/twelve/QuickstepCompatFactoryVS.java
@@ -44,8 +44,8 @@ public class QuickstepCompatFactoryVS extends QuickstepCompatFactoryVR {
             Class<?> remoteTransitionClass = Class.forName("android.window.RemoteTransition");
             Constructor<?> constructor =
                     remoteTransitionClass.getConstructor(
-                            IRemoteTransition.class, IApplicationThread.class);
-            return (RemoteTransition) constructor.newInstance(remoteTransition, appThread);
+                            IRemoteTransition.class, IApplicationThread.class, String.class);
+            return (RemoteTransition) constructor.newInstance(remoteTransition, appThread, debugName);
         } catch (ClassNotFoundException
                 | IllegalAccessException
                 | InstantiationException

